### PR TITLE
do search

### PR DESCRIPTION
**Pull request by seppo0010**

### CODE REVIEW

File: web/src/Document.tsx [hunk 1]
```diff
@@ -8,14 +8,16 @@ export interface DocumentData {
 }
 
 function Document(props: React.PropsWithChildren<DocumentData>) {
+  const [expanded, setExpanded] = useState(false);
+
   return (<Box>
     <div>
-      {props.content}
+      {props.content.slice(0, expanded ? undefined : 200)}
```
Comment (seppo0010) [inline web/src/Document.tsx:15]: extract magic number to a constant?

File: web/src/Document.tsx [hunk 2]
```diff
@@ -8,14 +8,16 @@ export interface DocumentData {
...
     </div>
     <div>
-      <Button>expand</Button>
+      <Button onClick={() => setExpanded(!expanded)} selected={expanded}>expand</Button>
```
Comment (seppo0010) [inline web/src/Document.tsx:18]: text should be in Spanish

File: web/src/Expediente.tsx [hunk 3]
```diff
@@ -15,77 +16,87 @@ export interface ExpedienteData {
...
-    fetch(`${process.env.PUBLIC_URL}/data/${props.file}.json`)
+    fetch(`${process.env.PUBLIC_URL}/data/${file}.json`)
       .then(res => res.json())
-      .then(json => {
-        setData(json)
-      })
+      .then(setData)
       .catch(e => {
-        setError(e.toString());
+        setError(error => error + 'DATA: ' + e.toString());
```
Comment (seppo0010) [inline web/src/Expediente.tsx:44]: wont this print "undefinedDATA"?

File: web/src/Document.tsx (hunk 2 above)
Comment (xaiki) [reply web/src/Document.tsx]: fixed

File: web/src/Document.tsx (hunk 1 above)
Comment (xaiki) [reply web/src/Document.tsx]: fixed

File: web/src/Button.tsx [hunk 4]
```diff
@@ -2,13 +2,18 @@ import React from 'react'
 import './Button.css'
 
 interface ButtonIface {
-  onClick?: React.MouseEventHandler<HTMLAnchorElement>
+  onClick?: Function
```
Comment (seppo0010) [inline web/src/Button.tsx:5]: `(ButtonIface) => void`?

File: web/src/Button.tsx [hunk 5]
```diff
@@ -2,13 +2,18 @@ import React from 'react'
...
   href?: string
   selected?: boolean
 }
 
-const Button = (props: React.PropsWithChildren<ButtonIface>) => {
-  return (<a className={`button ${props.selected ? 'selected' : null}`} {...props}>
+const Button = ({ selected, onClick, ...props }: React.PropsWithChildren<ButtonIface>) => {
+  const handleOnClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    onClick && onClick(props)
```
Comment (seppo0010) [inline web/src/Button.tsx:12]: send event?

File: web/src/Expediente.tsx [hunk 6]
```diff
@@ -14,35 +20,80 @@ export interface ExpedienteData {
...
-    <Section title={props.caratula}>
+    <Section title={<Highlighter text={caratula} term={search.term} />}>
       {
-        props.Actuaciones.map((a, i) => <Actuacion key={i} {...a} />)
+        actuaciones && actuaciones.map((a) => (
+          <Actuacion key={a.actId} data={a} search={search} />
+        ))
       }
-    </Section>
+    </ Section>
```
Comment (seppo0010) [inline web/src/Expediente.tsx:96]: accidental whitespace?

File: web/ts/create-index.ts [hunk 7]
```diff
@@ -1,17 +1,15 @@
-const MiniSearch = require('minisearch');
 import { promises as fs } from 'fs';
 import * as path from 'path';
+const MiniSearch = require('minisearch');
```
Comment (seppo0010) [inline web/ts/create-index.ts:3]: mixing import and require?

File: web/ts/create-index.ts [hunk 8]
```diff
@@ -1,17 +1,15 @@
...
+const MiniSearchConfig = require('../src/minisearch.config');
+
 const { StringDecoder } = require('string_decoder');
 const decoder = new StringDecoder('utf8');
 
-const DATADIR = '../../data';
+const DATADIR = '../public/data';
```
Comment (seppo0010) [inline web/ts/create-index.ts:9]: process.argv?

File: web/src/Expediente.tsx [hunk 9]
```diff
@@ -1,11 +1,17 @@
 import React, { useState, useEffect } from 'react';
 import Section from './Section';
 import Actuacion, { ActuacionData } from './Actuacion';
+import { SearchData } from './Search';
+import Highlighter from './Highlighter';
```
[the remark this answers is not in the document]
Comment (xaiki) [reply web/src/Expediente.tsx:5]: oops ,fixd

File: web/src/Button.tsx (hunk 4 above)
Comment (xaiki) [reply web/src/Button.tsx:5]: i'm not sure i get you ?

File: web/src/Button.tsx (hunk 5 above)
Comment (xaiki) [reply web/src/Button.tsx:12]: ?

File: web/src/Expediente.tsx (hunk 6 above)
Comment (xaiki) [reply web/src/Expediente.tsx]: fixd

File: web/ts/create-index.ts (hunk 7 above)
Comment (xaiki) [reply web/ts/create-index.ts:3]: for some reason it does not work with import. didn't want to burn time debugging.

File: web/ts/create-index.ts (hunk 8 above)
Comment (xaiki) [reply web/ts/create-index.ts:9]: then you'd need to maintain a shell script or a make file to run it...

File: web/src/Highlighter.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        <span className="no-hl">{length ? text.slice(0, length) : text}</span>
+      </React.Fragment>)
+    }
+
+    if (!term) {
+      updateFragment();
+      return;
+    };
+
+    const index = text.search(new RegExp(term, 'i'))
```
Comment (seppo0010) [inline web/src/Highlighter.tsx:24]: don't we have to escape "term" for regex?

**Review by seppo0010 — COMMENTED** [review]
btw, I cannot approve the pr since it was created by me, but I think it is good to go

File: web/src/Button.tsx (hunk 5 above)
Comment (seppo0010) [reply web/src/Button.tsx:12]: `onClick(props, event)`

File: web/src/Button.tsx (hunk 4 above)
Comment (seppo0010) [reply web/src/Button.tsx:5]: Function is too generic. Instead define the functions parameters and return type.

**Pull request merged**